map/map-test.js

```diff
@@ -473,18 +473,15 @@ QUnit.test("copying with assign() excludes special keys", function() {
 
 	var a = {
 		_data: {},
-		constructor: function() {},
-		__bindEvents: {},
 		"foo": "bar",
-		"existing": "newVal"
+		"existing": "neVal"
 	};
 
 	var b = new DefineMap({
 		"existing": "oldVal"
 	});
-	assign(b, a);
+	canReflect.assignMap(b, a);
 
-	QUnit.notEqual(a.constructor, b.constructor, "Constructor prop not copied");
```

Comment:
Was this removed because it shouldn’t have been tested in the first place?

Reply:
@matthewp and I removed it when we changed it to test assignments between two DefineMaps, as it wasn't deemed a realistic test.
However, since I changed it back to an assignment between a regular object and a DefineMap object it might be necessary to check against anyone attempting to assign a constructor to the DefineMap object.

Reply:
Adding the constructor test back in broke it, but I found a few solutions with @christopherjbaker that would fix it, but we should discuss it before they're implemented to see which one you prefer.